enterprise/cmd/frontend/internal/embeddings/resolvers/repo_embedding_jobs.go

```diff
@@ -0,0 +1,166 @@
+		return nil, err
+	}
+	total := int32(count)
+	return &total, nil
+}
+
+func (s *repoEmbeddingJobsConnectionStore) ComputeNodes(ctx context.Context, args *database.PaginationArgs) ([]graphqlbackend.RepoEmbeddingJobResolver, error) {
+	jobs, err := s.store.ListRepoEmbeddingJobs(ctx, args)
+	if err != nil {
+		return nil, err
+	}
+	resolvers := make([]graphqlbackend.RepoEmbeddingJobResolver, 0, len(jobs))
+	for _, job := range jobs {
+		resolvers = append(resolvers, &repoEmbeddingJobResolver{db: s.db, gitserverClient: s.gitserverClient, job: job})
+	}
+	return resolvers, nil
+}
+
+func (s *repoEmbeddingJobsConnectionStore) MarshalCursor(node graphqlbackend.RepoEmbeddingJobResolver, _ database.OrderBy) (*string, error) {
+	if node == nil {
+		return nil, errors.New("node is nil")
+	}
+	cursor := string(node.ID())
+	return &cursor, nil
+}
+
+func (s *repoEmbeddingJobsConnectionStore) UnmarshalCursor(cursor string, _ database.OrderBy) (*string, error) {
+	nodeID, err := unmarshalRepoEmbeddingJobID(graphql.ID(cursor))
+	if err != nil {
+		return nil, err
+	}
+	id := strconv.Itoa(nodeID)
+	return &id, nil
+}
+
+type repoEmbeddingJobResolver struct {
+	db              database.DB
+	gitserverClient gitserver.Client
+	job             *repobg.RepoEmbeddingJob
+}
+
+func (r *repoEmbeddingJobResolver) ID() graphql.ID {
+	return marshalRepoEmbeddingJobID(r.job.ID)
+}
+
+func (r *repoEmbeddingJobResolver) State() string {
+	return strings.ToUpper(r.job.State)
+}
+
+func (r *repoEmbeddingJobResolver) FailureMessage() *string {
+	return r.job.FailureMessage
+}
+
+func (r *repoEmbeddingJobResolver) QueuedAt() gqlutil.DateTime {
+	return gqlutil.DateTime{Time: r.job.QueuedAt}
+}
+
+func (r *repoEmbeddingJobResolver) StartedAt() *gqlutil.DateTime {
+	if r.job.StartedAt == nil {
+		return nil
+	}
+	return gqlutil.FromTime(*r.job.StartedAt)
+}
+
+func (r *repoEmbeddingJobResolver) FinishedAt() *gqlutil.DateTime {
+	if r.job.FinishedAt == nil {
+		return nil
+	}
+	return gqlutil.FromTime(*r.job.FinishedAt)
+}
+
+func (r *repoEmbeddingJobResolver) ProcessAfter() *gqlutil.DateTime {
+	if r.job.ProcessAfter == nil {
+		return nil
+	}
+	return gqlutil.FromTime(*r.job.ProcessAfter)
+}
+
+func (r *repoEmbeddingJobResolver) NumResets() int32 {
+	return int32(r.job.NumResets)
+}
+
+func (r *repoEmbeddingJobResolver) NumFailures() int32 {
+	return int32(r.job.NumFailures)
+}
+
+func (r *repoEmbeddingJobResolver) LastHeartbeatAt() *gqlutil.DateTime {
+	if r.job.ProcessAfter == nil {
+		return nil
+	}
+	return gqlutil.FromTime(*r.job.LastHeartbeatAt)
+}
+
+func (r *repoEmbeddingJobResolver) WorkerHostname() string {
+	return r.job.WorkerHostname
+}
+
+func (r *repoEmbeddingJobResolver) Cancel() bool {
+	return r.job.Cancel
+}
+
+func (r *repoEmbeddingJobResolver) Repo(ctx context.Context) (*graphqlbackend.RepositoryResolver, error) {
+	repo, err := r.db.Repos().Get(ctx, r.job.RepoID)
+	if err != nil {
+		// Skip resolving repository if it does not exist.
+		return nil, nil
+	}
```

Comment:
```suggestion
	if err != nil {
	if errcode.IsNotFound(err) {
		// Skip resolving repository if it does not exist.
		return nil, nil
		}
		return nil, err
	}
```
I think other errors should still be surfaced.